### PR TITLE
Fix #808

### LINKED ISSUE #808: IstioControlPlaneSpec Version validation pattern too wide
**Describe the bug**
The regex validating the version of IstioControlPlaneSpec.Version is too wide. The `.` should be escaped.

**Steps to reproduce the issue:**
Deploy operator chart version 2.0.5. Add a IstioControlPlane with
```yaml
spec:
   version: 12
```

**Expected behavior**
Validation fails in the k8s API.

### PR DESCRIPTION
**Pull request by Tanemahuta**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #808
| License         | Apache 2.0

### What's in this PR?
Improved matcher for supported istio versions

### Why?
IstioControlPlane was not usable setting 1.11 as version.

### Checklist
- [ X ] Implementation tested
- [ X ] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md) (N/A)
- [ X ] Logging code meets the guideline (N/A)
- [ X ] User guide and development docs updated (if needed) (N/A)

### CODE REVIEW

File: controllers/version.go [hunk 1]
```diff
@@ -19,10 +19,10 @@ package controllers
 import "regexp"
 
 const (
-	supportedIstioMinorVersionRegex = "^1.11"
+	supportedIstioMinorVersionRegex = "^1\\.11"
```
Comment (Laci21) [inline controllers/version.go]: What do you think about this? Just to account for sth like `1.11.4-rc.0`, but not allow sth like `1.11whatever`.

```suggestion
	supportedIstioMinorVersionRegex = "^1\\.11(\\.[0-9]+(?:-(?:dev|rc)\\.[0-9]+)?)?$"
```

Comment (Tanemahuta) [reply controllers/version.go]: What about `^1\\.11(\\.[0-9]+)?(-.+)?$` that should cover any any tag involving 1.11 (1.11.2, 1.11.2-test, 1.11-test, etc.) ?

Comment (Laci21) [reply controllers/version.go]: Sounds good as well, I guess it also matches `1.11.1-dev.` though.

Comment (Tanemahuta) [reply controllers/version.go]: Should cover every possible 1.11 semver tag. Modifying PR....

**Review by waynz0r — APPROVED** [review]
lgtm, thanks @Tanemahuta!

**Comment by Laci21** [comment]
Thank you very much @Tanemahuta for your contribution!

**Comment by Tanemahuta** [comment]
> Thank you very much @Tanemahuta for your contribution!

It was my pleasure, thanks for your help with the problems I had!

**Pull request merged**